packages/google_maps_flutter/google_maps_flutter_android/android/src/main/java/io/flutter/plugins/googlemaps/CozyMarkerBuilder.java

```diff
@@ -28,6 +30,28 @@ public class CozyMarkerBuilder {
         blankClusterMarker = getEmptyClusterBitmap(size);
         clusterTextStyle = getTextPaint(size / 3f, context);
         priceMarkerTextStyle = getTextPaint(size / 3.5f, context);
+        setMemoryCache();
+    }
+
+    private void setMemoryCache() {
```

Comment:
I think we should separate this whole Caching logic into a separate class for a better separation of responsibilities